engines/nancy/action/recordtypes.h

```diff
@@ -0,0 +1,411 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef NANCY_ACTION_RECORDTYPES_H
+#define NANCY_ACTION_RECORDTYPES_H
+
+#include "engines/nancy/action/actionrecord.h"
+#include "engines/nancy/commontypes.h"
+#include "engines/nancy/renderobject.h"
+
+#include "engines/nancy/commontypes.h"
+
+#include "common/stream.h"
+#include "common/array.h"
+#include "common/str.h"
+
+namespace Nancy {
+
+class NancyEngine;
+
+namespace Action {
+
+class SceneChange : public ActionRecord {
+public:
+    virtual void readData(Common::SeekableReadStream &stream) override;
+    virtual void execute() override;
+
+    SceneChangeDescription sceneChange;
+
+protected:
+    virtual Common::String getRecordTypeName() const override { return "SceneChange"; }
```

Comment:
I'm not sure how these objects are created - but giving the string "SceneChange" to the ctor of ActionRecord would most likely make this method being able to return a const ref to the string and also remove the virtual from the method.

Reply:
Yeah that could work, but then every instance of ActionRecord would need to lug around a string with its subclass name at all times. And since this is a debug function that's only used once in an object's lifetime (at creation), I don't know if there would be much benefit to this change.